src/AppBundle/Entity/Restaurant.php

```diff
@@ -722,4 +724,26 @@ public function addPreparationTimeRule($preparationTimeRule)
 
         return $this;
     }
+
+
+
+    /**
+     * @return mixed
+     */
+    public function getPledge()
+    {
+        return $this->pledge;
+    }
+
+    /**
+     * @param mixed $pledge
+     *
+     * @return self
+     */
+    public function setPledge($pledge)
+    {
+        $this->pledge = $pledge;
```

Comment:
Actually, the linked restaurant isn't saved in database. 
You need to add `$pledge->setRestaurant($this);` here.